In [3]:
from Bio import Entrez
entrez_email = raw_input("What is an e-mail you would like to provide to query the Entrez database?    ")
Entrez.email = entrez_email
gene_entrez_id = raw_input('Enter Gene Entrez ID:')
handle = Entrez.efetch(db="gene", id=gene_entrez_id ,rettype="fasta", retmode="xml")
records=Entrez.read(handle)
#records[0].keys()
gene_entrez_name = records[0]["Entrezgene_gene"]["Gene-ref"]["Gene-ref_locus"]
gene_entrez_location = records[0]["Entrezgene_gene"]["Gene-ref"]["Gene-ref_maploc"]
gene_entrez_syn = records[0]["Entrezgene_gene"]["Gene-ref"]["Gene-ref_syn"]
gene_entrez_summary=records[0]["Entrezgene_summary"]
print "\nName: %s\n\nLocation: %s\n\nSynonyms: %s \n\nSummary: %s" %(gene_entrez_name, gene_entrez_location, gene_entrez_syn, gene_entrez_summary)
import os 
current_fp = os.getcwd()

What is an e-mail you would like to provide to query the Entrez database?    lac018@ucsd.edu
Enter Gene Entrez ID:445

Name: ASS1

Location: 9q34.11

Synonyms: ['ASS', 'CTLN1'] 

Summary: The protein encoded by this gene catalyzes the penultimate step of the arginine biosynthetic pathway. There are approximately 10 to 14 copies of this gene including the pseudogenes scattered across the human genome, among which the one located on chromosome 9 appears to be the only functional gene for argininosuccinate synthetase. Mutations in the chromosome 9 copy of this gene cause citrullinemia. Two transcript variants encoding the same protein have been found for this gene. [provided by RefSeq, Aug 2012]


#### Refseq ID for ARGSS/ASS1 (Gene Entrez ID: 445) is  NP_446464 according to GEMPRO

In [4]:
acc = raw_input('Enter the refseq ID (u_refseq) of the sequence you would like to download:   ')
from Bio import SeqIO
from Bio import Entrez
Entrez.email = entrez_email
temp = Entrez.efetch(db="nucleotide",rettype="gb",id=acc)
out = open(acc+".faa",'w')
gbseq = SeqIO.read(temp, "genbank")
SeqIO.write(gbseq,out,"fasta")
temp.close()
out.close()
print(gbseq)

Enter the refseq ID (u_refseq) of the sequence you would like to download:    NP_446464


HTTPError: HTTP Error 400: Bad Request

#### Example with different refseq ID

For example: NP_000041.2  (also for ASS1), taken from https://www.ncbi.nlm.nih.gov/gene/445. It seems like some refseq IDs give the same error as above while others work

In [7]:
acc = raw_input('Enter the refseq ID (u_refseq) of the sequence you would like to download:   ')
from Bio import SeqIO
from Bio import Entrez
Entrez.email = entrez_email
temp = Entrez.efetch(db="nucleotide",rettype="gb",id=acc)
out = open(acc+".faa",'w')
gbseq = SeqIO.read(temp, "genbank")
SeqIO.write(gbseq,out,"fasta")
temp.close()
out.close()
print(gbseq)

Enter the refseq ID (u_refseq) of the sequence you would like to download:   NM_000050.4
ID: NM_000050.4
Name: NM_000050
Description: Homo sapiens argininosuccinate synthase 1 (ASS1), transcript variant 1, mRNA.
Number of features: 31
/comment=REVIEWED REFSEQ: This record has been curated by NCBI staff. The
reference sequence was derived from DB496935.1, AL354898.10 and
BC013224.1.
This sequence is a reference standard in the RefSeqGene project.
On Aug 25, 2006 this sequence version replaced gi:53759106.
Summary: The protein encoded by this gene catalyzes the penultimate
step of the arginine biosynthetic pathway. There are approximately
10 to 14 copies of this gene including the pseudogenes scattered
across the human genome, among which the one located on chromosome
9 appears to be the only functional gene for argininosuccinate
synthetase. Mutations in the chromosome 9 copy of this gene cause
citrullinemia. Two transcript variants encoding the same protein
have been found for this gene